Docs/BazelForiOSDevelopers.md

```diff
@@ -0,0 +1,370 @@
+    bundle_id = "com.bazel-bootcamp.some",
+    families = ["iphone"],
+    infoplists = ["Info.plist"],
+    minimum_os_version = "[MINIMUM VERSION]",
+    # Add `sources` as a dependency
+    deps = [ ":sources" ],
+)
+```
+
+The`ios_application` and `objc_library` targets together would be represented
+in Xcode as:
+
+![Docs](XcodeExampleOfiOSProject.png)
+
+### Rules
+
+In the previous segment, we created an iOS application with a single BUILD
+file. The `ios_application` rule is implemented by `rules_apple` and the
+`objc_library` is a native rule.
+
+A target is an instance of a rule. Rules implement business logic for how the
+iOS application is built by creating actions. actions represent invocations of
+extenal command line programs like `clang` or `bash`. Typical projects contain
+many rules, targets, and BUILD files.
+
+```
+BUILD file -> target -> rule -> action -> execution
+```
+
+Bazel and open source rules should provide most functionality to build an iOS
+application. Generally, defining custom rules isn't required but can improve
+and consolidate functionality. To learn more about how to create custom rules,
+the document, [Bazel extensions](https://docs.bazel.build/versions/master/skylark/concepts.html)
+contains a comprehensive overview.
+	
+### Command line usage
+
+In Xcode, command line builds are achieved through the command line interface
+`xcodebuild`.  to build the scheme `ios-app` from `MyProject.xcworkspace`,
+`xcodebuild`, would be invoked as:
+
+```
+xcodebuild -workspace MyProject.xcworkspace -scheme ios-app
+```
+
+Bazel exposes a command line which can also do builds. In Bazel, _every_ target
+is in the global WORKSPACE and there is no notion of schemes.
+```
+bazel build ios-app
+```
+
+The Bazel command line has hundreds of options, which can be found in the
+[Bazel
+documentation](https://docs.bazel.build/versions/master/command-line-reference.html).
+For iOS developers, a set of useful flags is available at
+[bazel-ios-users](https://github.com/ios-bazel-users/ios-bazel-users/blob/master/UsefulFlags.md).
+
+In addition to the default bazel options, it's common to create custom
+configuration settings to customize builds, so the possible Bazel command line
+permutations are endless.
+
+For example, the following build file has conditional `copts` on `:app_store`
+
+```
+objc_library(
+   name = "some",
+   srcs = ["some.m"],
+   copts = select({
+      # For app_store we need to build `some.m` with a special copt
+      ":app_store": ["-DAPPSTORE=1"],
+      "//conditions:default": [],
+    })
+)
+config_setting(
+    name = "app_store",
+    values = { "app_store" : "true" }
+)
+```
+
+This is passed in as a define to Bazel
+```
+bazel build ios-app --define app_store=true
+```
+
+For more information about `config_setting`, please see the [Bazel
+documentation](https://docs.bazel.build/versions/master/be/common-definitions.html#configurable-attributes).
+
+### Configuration of libraries and flags via macros
```

Comment:
I think it might be simpler to title this section "Macros" similar to "Rules"

Reply:
Will do! I've got a segment for `#rules` as well, so I'll scope down to macros